private/syntax/compile/binding-spec.rkt

```diff
@@ -53,16 +55,20 @@
 (struct ref [pvar] #:transparent)
 (struct bind with-stx [pvar] #:transparent)
 (struct bind-syntax with-stx [pvar transformer-pvar] #:transparent)
-(struct bind-syntaxes with-stx [pvar transformer-pvar] #:transparent)
-(struct rec with-stx [pvars] #:transparent)
-(struct re-export with-stx [pvars] #:transparent)
+(struct bind-syntaxes with-stx [depth pvar transformer-pvar] #:transparent)
+(struct rec with-stx [depth pvar] #:transparent)
+; no surface syntax, just a mechanism to combine recs
+; something like [(import x) (import y)] ~> (recs (list (rec x) (rec y)))
+(struct recs [specs] #:transparent)
```

Comment:
The field of recs could simply contain pvars; they'll get there from the pass that combines together `import` forms into a single `recs` form, which will happen after things like depth checking.